### ISSUE TITLE
Compute SPI on 1D timeseries

### ISSUE THREAD

**Issue by tommylees112** [monocongo/climate_indices#306]
**Is your feature request related to a problem? Please describe.**
I was hoping to produce an SPI from my timeseries, however, the scripts only seem to work with multi-dimensional inputs.

```
In[]:
process_climate_indices --index spi --periodicity monthly --netcdf_precip ~/Downloads/ichirps_20_25_Kenya_19812010a.txt.nc --var_name_precip  pr --output_file_base ~/Downloads/nclimgrid_lowres --scales 6 12 --calibration_start_year 1981 --calibration_end_year 2010 --multiprocessing all

Out[]:
2019-04-26  14:37:15 INFO Start time:    2019-04-26 14:37:15.913817
2019-04-26  14:37:16 ERROR Invalid dimensions of the precipitation variable: ('time',)
Valid dimension names and order: [('lat', 'lon', 'time'), ('time', 'lat', 'lon'), ('time', 'division'), ('division', 'time')]
2019-04-26  14:37:16 ERROR Failed to complete
Traceback (most recent call last):
  File "/Users/tommylees/miniconda3/envs/crop/lib/python3.7/site-packages/climate_indices/__main__.py", line 1644, in main
    input_type = _validate_args(arguments)
  File "/Users/tommylees/miniconda3/envs/crop/lib/python3.7/site-packages/climate_indices/__main__.py", line 132, in _validate_args
    raise ValueError(msg)
ValueError: Invalid dimensions of the precipitation variable: ('time',)
Valid dimension names and order: [('lat', 'lon', 'time'), ('time', 'lat', 'lon'), ('time', 'division'), ('division', 'time')]
Traceback (most recent call last):
  File "/Users/tommylees/miniconda3/envs/crop/bin/process_climate_indices", line 11, in <module>
    sys.exit(main())
  File "/Users/tommylees/miniconda3/envs/crop/lib/python3.7/site-packages/climate_indices/__main__.py", line 1644, in main
    input_type = _validate_args(arguments)
  File "/Users/tommylees/miniconda3/envs/crop/lib/python3.7/site-packages/climate_indices/__main__.py", line 132, in _validate_args
    raise ValueError(msg)
ValueError: Invalid dimensions of the precipitation variable: ('time',)
Valid dimension names and order: [('lat', 'lon', 'time'), ('time', 'lat', 'lon'), ('time', 'division'), ('division', 'time')]
```

**Describe the solution you'd like**
I want to pass a 1D timeseries and return an output with the same dimensions.

```
<xarray.Dataset>
Dimensions:  (time: 14244)
Coordinates:
  * time     (time) datetime64[ns] 1981-01-01 1981-01-02 ... 2019-12-31
Data variables:
    pr       (time) float32 ...
Attributes:
    title:                      spatial statistic of CHIRPS-2.0 merged satell...
    description:                CHIRPS precipitation Kenya
    file:                       ./data/ichirps_20_25_Kenya.dat
    minimal_valid_fraction:      30.00
    olderfile:                  CHIRPSData/v2p0chirps_25.nc
    cdi:                        Climate Data Interface version ?? (http://mpi...
    source:                     ftp://chg-ftpout.geog.ucsb.edu/pub/org/chg/pr...
    gdal:                       GDAL 2.2.3, released 2017/11/20
    nco:                        4.7.2
    cdo:                        Climate Data Operators version 1.9.3 (http://...
    geospatial_lat_max:         40.0
    geospatial_lon_min:         -20.0
    geospatial_lon_units:       degrees_east
    geospatial_lon_resolution:  0.2
    time_coverage_start:        1981-01-01
    time_coverage_end:          2019-03-31
    climexp_url:                https://climexp.knmi.nl/select.cgi?chirps_20_25
    mask:                       data/mask_Kenya_chirps_20_25.nc
    institution:                KNMI Climate Explorer
    comment:                    \n
    history:                     2019-04-26 15:31:05 bin/dat2nc data/ichirps_...
    Conventions:                CF-1.0
```

**Describe alternatives you've considered**
I considered just adding lat/lon to the .nc file but that also failed:

```
In[]:
d['lat'] = [1]
d['lon'] = [1]
d.to_netcdf('~/Downloads/kenya_chirps.nc')
d

Out[]:
<xarray.Dataset>
Dimensions:  (lat: 1, lon: 1, time: 14244)
Coordinates:
  * time     (time) datetime64[ns] 1981-01-01 1981-01-02 ... 2019-12-31
  * lat      (lat) int64 1
  * lon      (lon) int64 1
Data variables:
    pr       (time) float32 ...
Attributes:
    title:                      spatial statistic of CHIRPS-2.0 merged satell...
    description:                CHIRPS precipitation Kenya
    file:                       ./data/ichirps_20_25_Kenya.dat
    minimal_valid_fraction:      30.00
    olderfile:                  CHIRPSData/v2p0chirps_25.nc
    cdi:                        Climate Data Interface version ?? (http://mpi...
    source:                     ftp://chg-ftpout.geog.ucsb.edu/pub/org/chg/pr...
    gdal:                       GDAL 2.2.3, released 2017/11/20
    nco:                        4.7.2
    cdo:                        Climate Data Operators version 1.9.3 (http://...
    geospatial_lat_max:         40.0
    geospatial_lon_min:         -20.0
    geospatial_lon_units:       degrees_east
    geospatial_lon_resolution:  0.2
    time_coverage_start:        1981-01-01
    time_coverage_end:          2019-03-31
    climexp_url:                https://climexp.knmi.nl/select.cgi?chirps_20_25
    mask:                       data/mask_Kenya_chirps_20_25.nc
    institution:                KNMI Climate Explorer
    comment:                    \n
    history:                     2019-04-26 15:31:05 bin/dat2nc data/ichirps_...
    Conventions:                CF-1.0

```

**Additional context**

**Comment by monocongo** [comment]
Thanks for bringing this to my attention, @tommylees112. Yes, this is a novel use case, but I have recently used the code to perform indices computation on 1-D timeseries (see this [Jupyter notebook](https://github.com/monocongo/climate_indices/blob/master/notebooks/PDSI_Canadian_Stations.ipynb)). If you want to send me a link to your dataset I will try to cook up a similar notebook and/or script for your use case, it should be reasonably straight-forward. Or have a go at it yourself, following the above-mentioned notebook as an example, and please feel free to contribute it into this repo via a pull request, it will be very appreciated.

Maybe you can send me the full information from the `source` attribute so I can fetch the data? I have poked around on the CHIRPS page but I can't find what I'm looking for (daily precipitation for Kenya, right?), and the above source info (`ftp://chg-ftpout.geog.ucsb.edu/pub/org/chg/pr...`) isn't enough to help. With access to the data it will be much easier for me to add 1-D support into the script endpoint and/or a new notebook. I could cook up an example dataset myself, but I'm lazy and it's not certain to match your requirements.

**Comment by tommylees112** [comment]
The data is from the KNMI climate explorer: https://climexp.knmi.nl/

You can get the data from this link:
```
wget --header 'Host: climexp.knmi.nl' --user-agent 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.14; rv:65.0) Gecko/20100101 Firefox/65.0' --header 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8' --header 'Accept-Language: en-GB,en;q=0.5' --referer 'https://climexp.knmi.nl/dat2nc.cgi?datafile=data/ichirps_20_25_Kenya.dat&type=i&station=CHIRPS_precipitation_Kenya&id=4a6b3503812540ea4e37773f2cc21cd8' --header 'Upgrade-Insecure-Requests: 1' 'https://climexp.knmi.nl/data/ichirps_20_25_Kenya.nc' --output-document 'ichirps_20_25_Kenya.nc'
```

Thank you @monocongo ! I will have a go with the notebook and let you know how it goes.
Tommy

**Comment by monocongo** [comment]
@tommylees112 I have added code that appears to work now for 1-D/timeseries NetCDF inputs when computing daily SPI at 30-day scale on the CHIRPS Kenya dataset you referenced. Please give this a try. The branch name is `issue_306_1d_timeseries_netcdf`.

I ran the code using the following command:
```
$ python /home/james/git/climate_indices/climate_indices/__main__.py --index spi --scales 30 --netcdf_precip /home/james/data/climate_indices_306/ichirps_20_25_Kenya.nc --var_name_precip pr --periodicity daily --output_file_base /home/james/data/climate_indices_306/ichirps_20_25_Kenya --calibration_start_year 1984 --calibration_end_year 2016 --multiprocessing all
```
I haven't tested to make sure that these changes don't introduce new issues for previously working processing use cases so I haven't merged this into the master branch yet. Please let me know if this is working for you and if the resulting SPI data is looking reasonable. Thanks!

**Comment by monocongo** [comment]
![ichirps_20_25_Kenya_spi_gamma_30](https://user-images.githubusercontent.com/1328158/56939535-e8a23180-6ad6-11e9-945c-418da6f1a213.png)

**Comment by monocongo** [comment]
![ichirps_20_25_Kenya_spi_pearson_](https://user-images.githubusercontent.com/1328158/56939589-2acb7300-6ad7-11e9-96c4-f8ba45bb5064.png)

**Comment by tommylees112** [comment]
Hi @monocongo thanks very much that works a treat for me! 

```
python climate_indices/__main__.py --index spi --scales 30 --netcdf_precip /Users/tommylees/github/grun_drought/data/raw/ichirps_20_25_Kenya.nc --output_file_base /Users/tommylees/github/grun_drought/data/interim/Kenya_SPI --calibration_start_year 1984 --calibration_end_year 2016 --multiprocessing all --periodicity daily --var_name_precip pr

2019-04-30  11:06:25 INFO Start time:    2019-04-30 11:06:25.606093
2019-04-30  11:06:25 INFO Computing 30-day SPI/Pearson
2019-04-30  11:06:36 INFO Computing 30-day SPI/Gamma
2019-04-30  11:06:40 INFO End time:      2019-04-30 11:06:40.895767
2019-04-30  11:06:40 INFO Elapsed time:  0:00:15.289674
```

![spi_test](https://user-images.githubusercontent.com/21049064/56955416-38194980-6b39-11e9-9904-8ce54e4b464a.png)

**Comment by monocongo** [comment]
Resolved by PR #319
